### PR TITLE
Fix bugs windows

### PR DESCRIPTION
**Pull request by denisenkom**
Fix for issue #59

### CODE REVIEW

File: ykcs11/utils.c [hunk 1]
```diff
@@ -76,7 +76,7 @@ CK_RV parse_readers(ykpiv_state *state, const CK_BYTE_PTR readers, const CK_ULON
         }
       }
       (*n_slots)++;
-      p += i + 1;
+      p = &readers[i] + 1;
```
Comment (a-dma) [inline ykcs11/utils.c]: Good catch.
Very minor nit, I would prefer `readers + i + 1`

File: ykcs11/ykcs11.c [hunk 2]
```diff
@@ -51,7 +51,7 @@ CK_DEFINE_FUNCTION(CK_RV, C_Initialize)(
 )
 {
   CK_BYTE readers[2048];
-  CK_ULONG len = sizeof(readers);
+  size_t len = sizeof(readers);
```
Comment (a-dma) [inline ykcs11/ykcs11.c]: Also, good catch.

File: ykcs11/ykcs11.c [hunk 3]
```diff
@@ -536,6 +536,7 @@ CK_DEFINE_FUNCTION(CK_RV, C_OpenSession)(
     session.info.state = CKS_RO_PUBLIC_SESSION; // Nobody has logged in, default RW session
   }
 
+  session.info.slotID = slotID;
```
Comment (a-dma) [inline ykcs11/ykcs11.c]: This is already at line 525 but is commented out. We might as well uncomment that line.

**Comment by a-dma** [comment]
Hey, thanks for the fixes, these things are annoying to debug.

I've left a few comments, in general I want to merge the change to these two files (utils.c and ykcs11.c).
What I'm not so keen on merging is the part regarding the consolidation of the header files. I prefer to have two separate files one for types and one with the rest.

File: ykcs11/ykcs11.c (hunk 3 above)
Comment (denisenkom) [inline ykcs11/ykcs11.c]: Looks different to me, this one updates session info, while the one at 525 seems to update slot info.

**Comment by denisenkom** [comment]
Yeah, I am not sure about consolidation too.  On one hand modifying third-party files will be a pain to support.  Another option is to use different pkcs11.h file: https://github.com/OpenSC/OpenSC/blob/master/src/pkcs11/pkcs11.h
it already supports Windows.

**Comment by a-dma** [comment]
It seems to be working fine now so unless we have a good reason I'm inclined to keep it as it is for the time being.

Anyway if you want I'm happy do keep the discussion going, but I would suggest to move it to its own issue and keep this PR just for the Win64 stuff so I can merge that in.

**Comment by denisenkom** [comment]
I totally agree!  Let's merge.

**Comment by a-dma** [comment]
Thanks!

**Pull request merged**